src/Spice86.Core/Emulator/OperatingSystem/DosFileManager.cs

```diff
@@ -602,119 +556,57 @@ public class DosFileManager {
         return (uint)newOffset;
     }
 
-    private void SetOpenFile(ushort fileHandle, OpenFile? openFile) {
-        _openFiles[fileHandle] = openFile;
+    private void SetOpenFile(ushort fileHandle, OpenFile? openFile) => OpenFiles[fileHandle] = openFile;
+    
+    private void UpdateDosTransferAreaFromFile(string matchingFile) {
+        if (_loggerService.IsEnabled(Serilog.Events.LogEventLevel.Verbose)) {
+            _loggerService.Verbose("Found matching file {MatchingFile}", matchingFile);
+        }
+        DosDiskTransferArea dosDiskTransferArea = new(_memory, GetDiskTransferAreaAddressPhysical());
+        FileInfo attributes = new FileInfo(matchingFile);
+        DateTime creationZonedDateTime = attributes.CreationTimeUtc;
+        DateTime creationLocalDate = creationZonedDateTime.ToLocalTime();
+        DateTime creationLocalTime = creationZonedDateTime.ToLocalTime();
+        dosDiskTransferArea.FileDate = ToDosDate(creationLocalDate);
+        dosDiskTransferArea.FileTime = ToDosTime(creationLocalTime);
+        dosDiskTransferArea.FileSize = (ushort)attributes.Length;
+        dosDiskTransferArea.FileName = Path.GetFileName(matchingFile);
     }
 
-    private string? ToCaseSensitiveFileName(string? caseInsensitivePath) {
-        if (string.IsNullOrWhiteSpace(caseInsensitivePath)) {
-            return null;
+    /// <summary>
+    /// Creates a directory on disk.
+    /// </summary>
+    /// <param name="directory">The directory name to create</param>
+    /// <returns></returns>
+    /// <returns>A <see cref="DosFileOperationResult"/> with details about the result of the operation.</returns>
+    public DosFileOperationResult CreateDirectory(string directory) {
+        string hostPath = _dosPathResolver.PrefixWithHostDirectory(directory);
+
+        string? fullPath = _dosPathResolver.ToHostCaseSensitiveFullName(directory, true);
```

Comment:
Not sure to understand why converting the param in input and not hostpath

Reply:
I moved hostPath closer to where it's used.

hostPath is for creation of the folder, fullPath is for verification if we can do it.